### PR TITLE
Improve typing of the drop_frame parameter on the RationalTime.to_timecode method

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
**Summarize your change.**

A couple of months ago, I noticed that the `drop_frame` parameter of the `RationalTime.to_timecode` method was using `py::object` as its type. This is confusing for the user.

With the change, running `help(opentimelineio.opentime.RationalTime.to_timecode)` returns:

```
to_timecode(...)
    to_timecode(*args, **kwargs)
    Overloaded function.
    
    1. to_timecode(self: opentimelineio._opentime.RationalTime, rate: float, drop_frame: Optional[bool]) -> str
```

And it still works as before:
```python
>>> opentimelineio.opentime.RationalTime(30 * 59 + 30).to_timecode(29.97, 0)
'00:29:58:06'
>>> opentimelineio.opentime.RationalTime(30 * 59 + 30).to_timecode(29.97, False)
'00:29:58:06'
>>> opentimelineio.opentime.RationalTime(30 * 59 + 30).to_timecode(29.97, 1)
'00:30:00;00'
>>> opentimelineio.opentime.RationalTime(30 * 59 + 30).to_timecode(29.97, True)
'00:30:00;00'
>>> opentimelineio.opentime.RationalTime(30 * 59 + 30).to_timecode(29.97, None)
'00:30:00;00'
```

### CODE REVIEW

**Review by ssteinbach — COMMENTED** [review]
Looks pretty good, just wanted to make sure that I understand the new header and how it ties into this change.

File: src/py-opentimelineio/bindings-common/casters.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using nonstd::optional;
+using nonstd::nullopt_t;
+
+namespace pybind11 { namespace detail {
+    template<typename T> struct type_caster<optional<T>>
+        : public optional_caster<optional<T>> {};
+
+    template<> struct type_caster<nullopt_t>
+        : public void_caster<nullopt_t> {};
+}}
```
[the remark this answers is not in the document]
Comment (JeanChristopheMorinPerso) [reply src/py-opentimelineio/bindings-common/casters.h:14]: Yes. It's the documented way to support optionals. `void_caster` and `optional_caster` are defined in pybind11. Once we move to C++17, these custom casters won't be needed anymore. I'll add a comment.

File: src/py-opentimelineio/bindings-common/casters.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/operators.h>
+#include "nonstd/optional.hpp"
+
+using nonstd::optional;
+using nonstd::nullopt_t;
+
+namespace pybind11 { namespace detail {
+    template<typename T> struct type_caster<optional<T>>
```
[the remark this answers is not in the document]
Comment (JeanChristopheMorinPerso) [reply src/py-opentimelineio/bindings-common/casters.h:13]: I added a comment with a link to the Pybind11 documentation. Let me know if it's enough.

[the remark this answers is not in the document]
Comment (ssteinbach) [reply src/py-opentimelineio/bindings-common/casters.h:13]: Perfect, thanks!

**Review by ssteinbach — APPROVED** [review]

**Pull request merged**